src/main/java/org/lightcouch/CouchDbClient.java

```diff
@@ -95,6 +98,41 @@
  */
 public class CouchDbClient extends CouchDbClientBase {
 
+    //User-Agent value for the client
+    public static final String USER_AGENT;
+
+    static {
+        String ua = "java-cloudant";
+        String version = "unknown";
+        final URL url = CouchDbClient.class.getClassLoader().getResource("client.properties");
+        final Properties properties = new Properties();
+        InputStream propStream = null;
+        try {
+            properties.load((propStream = url.openStream()));
+            ua = properties.getProperty("user.agent.name", ua);
+            version = properties.getProperty("user.agent.version", version);
+        } catch (Exception ex) {
+            //swallow exception and keep using default values
+        } finally {
+            if (propStream != null) {
+                try {
+                    propStream.close();
+                } catch (IOException e) {
+                    //can't do anything else
```

Comment:
Is it worth trying to add an error message in the response if an exception occurs?